src/communication/websocket/session.cpp

```diff
@@ -173,4 +211,21 @@ void Session::OnRead(const boost::beast::error_code ec, const size_t /*bytes_tra
   DoRead();
 }
 
+bool Session::IsAuthenticated() const noexcept { return authenticated_ || !auth_.HasAnyUsers(); }
```

Comment:
`auth_.HasAnyUsers()` is not `noexcept`, so I think this function also cannot be `noexcept`.